sdk/releases.md

```diff
@@ -0,0 +1,36 @@
+# Ably SDK Team: Guidance on Releases
+
+This document outlines our ideal requirements for SDK releases.
+
+There will be some SDK repositories:
+
+- which are yet to have the work done on them which will implement this
+- where it's necessary to deviate from some of the ideals specified here due to constraints which are innate or otherwise idiomatic to the platform being released for
+
+Where technically possible, SDK repository release processes which claim to adhere to this guidance must follow these ideal requirements closely.
+
+## Release Process
+
+1. Merge all pull requests containing changes intended for this release to `main` branch
+2. Prepare a [Release Branch](#release-branch) and a corresponding pull request, obtain approval from reviewers and then merge to `main` branch
+3. Trigger the [Publish Workflow](#publish-workflow)
+4. If the publish fails then add commits to the `main` branch that aim to fix the problem and then try step 3 again
```

Comment:
I'm not sure how I feel about direct pushes to `main`, I think we should always encourage changes to be peer reviewed, or at least a PR + merge with an explicit comment justifying why the change was merged without a review.

Reply:
I agree - branch protection should be in place so this isn't possible

Reply:
Thanks, both. I worded that really badly. I would prefer that these went via pull requests and should have made that clearer. I'll work on improving this phrasing.

Reply:
https://github.com/ably/engineering/pull/23/commits/5a0f0d8236d2b215cea105e3b3a6c3b6d5679634